notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,567 @@
+
+Recall that the span of $X$ is important because it is the subspace we want to project $\vec{y}$ onto. If the subspace does not change, then the projection will not change.
+
+For example, when we introduced $\vec{x}$ when moving from the constant model to linear model, we introduced a independent variable: $\vec{x} = \begin{bmatrix} 3 \\ 0 \\ -1 \end{bmatrix}$ cannot be expressed as a scalar of $\begin{bmatrix} 1 \\ 1 \\ 1 \end{bmatrix}$.
+
+However, say we instead introduced a new variable $d$ with values:
+
+| d | y |
+| - | :-|
+| -0.5 | 2 |
+| -0.5 | 1 |
+| -0.5 | -2 |
+
+Then, we would be minimizing the loss function:
+
+$$ 
+\begin{aligned}
+L(\hat{\theta}, \vec{d}, \vec{y})
+&= \left \Vert  \qquad   
+\begin{bmatrix} y_1 \\ y_2 \\ y_3  \end{bmatrix} \quad - \quad 
+\begin{bmatrix} 1 & d_1 \\ 1 & d_2 \\ 1 & d_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad   
+\begin{bmatrix} y_1 \\ y_2 \\ y_3  \end{bmatrix} \quad - \quad 
+\begin{bmatrix} 1 & -0.5 \\ 1 & -0.5 \\ 1 & -0.5 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+X 
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+\hat{y}
+\qquad \right \Vert ^2 \\
+\end{aligned}
+$$
+
+Recall that $\hat{y}$ can take on any of the infinite linear combinations of the columns of $X$. Thus, its possible values are $\theta_0 \begin{bmatrix} 1 \\ 1 \\ 1 \end{bmatrix} + \theta_1 \begin{bmatrix} -0.5 \\ -0.5 \\ -0.5 \end{bmatrix}$. 
+
+
+Since $\vec{d} = \frac{-1}{2} \ \vec{1}$, regardless of $\theta_0$ and $\theta_1$, the possible values can be rewritten as:
+$$\theta_0 \begin{bmatrix} 1 \\ 1 \\ 1 \end{bmatrix} - \frac{1}{2} \theta_1 \begin{bmatrix} 1 \\ 1 \\ 1 \end{bmatrix} \quad = \quad (\theta_0 - \frac{1}{2} \theta_1) \begin{bmatrix} 1 \\ 1 \\ 1 \end{bmatrix}$$
+
+Just as before, our solution space are the scalar multiples of $\vec{1}$, just like before we introduced the variable $d$. 
+
+<img src="../../notebooks-images/linear_projection__dependent_variables.png" width="400" />
+
+
+Adding a linearly dependent column to $X$ does not change its spanning column space, and thus will not change the projection and solution to the least squares problem.
+
+## Vector Space Review
```

Comment:
Wow I can't believe you added this whole section.. you are insane :O 

Do you think this belongs in the same section O_O

Reply:
lol yea spent too long this week haha. yea will need discussion